File: 7-kyu-kata/factorial-1/python/factorial_1_test.py
# 7 kyu - Factorial  [ ID: 57a049e253ba33ac5e000212  (factorial-1) ]
# URL: https://www.codewars.com/kata/57a049e253ba33ac5e000212
# Category: REFERENCE  |  Tags: FUNDAMENTALS
# ******************************************************************************
import codewars_test as test
from solution import factorial


@test.describe("Fixed Tests")
def fixed_tests():
    @test.it("Basic Test Cases")
    def basic_test_cases():

        tests = (
            (0, 1),
            (1, 1),
            (2, 2),
            (3, 6),
            (4, 24),
            (5, 120),
            (6, 720),
            (7, 5040),
        )

        for t in tests:
            inp, exp = t
            test.assert_equals(factorial(inp), exp)


@test.describe("Random Tests")
def _():

    from random import randint
    from math import factorial as _reference

    for _ in range(100):
        n = randint(0, 17)

        @test.it(f"testing for factorial({n})")
        def test_case():
            test.assert_equals(factorial(n), _reference(n))

File: 7_Kyu_Kata/factorial-1/JavaScript/factorial1.Test.js
// [object Object] - Factorial  [ ID: 57a049e253ba33ac5e000212  (factorial-1) ]
// URL: https://www.codewars.com/kata/57a049e253ba33ac5e000212
// Category: REFERENCE  |  Tags: FUNDAMENTALS
// *****************************************************************************
describe("Basic tests",() =>{
Test.assertEquals(factorial(0), 1);
Test.assertEquals(factorial(1), 1);
Test.assertEquals(factorial(4), 24);
Test.assertEquals(factorial(7), 5040);
Test.assertEquals(factorial(17), 355687428096000);
})

describe("Random tests",() =>{
const randint=(a,b)=>~~(Math.random()*(b-a+1)+a);
const sol=n=>Array.from({length:n},(a,i)=>i+1).reduce((a,b)=>a*b,1);

for (let i=0;i<40;i++){
  let n=randint(0,17);
  it(`Testing for ${n}`,() =>{
  Test.assertEquals(factorial(n), sol(n), "It should work for random inputs too");
  })
}
})
